lib/cucumber/core/test/case.rb

```diff
@@ -61,6 +62,7 @@ def match_locations?(queried_locations)
 
         def matching_locations
           [
+            parent_locations,
```

Comment:
this is the thing you explained to me iirc where you said this is returning an array with a load of lines, and so long as it's "somewhere" in here we're happy right?

Reply:
@luke-hill Yes, exactly. Its essentially a order-independent set of lines.

So maybe it makes sense to refactor it to be an actual `Set` rather than an order-dependent `Array`?

Reply:
:shrug: for now. You can decide. For order independent things I usually lean more towards hash as they're self descriptive when it comes to refactoring. It's an option I do in my own gem here: https://github.com/site-prism/site_prism/blob/main/lib/site_prism/dsl/builder.rb#L23

Reply:
@luke-hill I'll try a little spike of refactoring to Set. I think it'll be nice and contained, and can nicely resolve the other `[0]` issue, too. We'll see.

Reply:
@luke-hill Hmm, yeah, refactoring to a Set didn't really add anything except a bit more incidental complexity. I vote to leave it as-is.